spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -114,6 +115,12 @@ private Blob resolve() throws IOException {
 	}
 
 	private Blob createBlob() throws IOException {
+		// Creating the bucket resource will create the bucket if necessary.
+		Resource bucket = getBucket();
+		if (!bucket.exists()) {
```

Comment:
I'd prefer to see another kind of logic here, like -- check if the bucket exists, if it doesn't, create it if you can (e.g. configuration allows) by calling an explicit `create()`-like method (not a constructor).